server/server.go

```diff
@@ -1,35 +1,46 @@
 package server
 
 import (
+	"database/sql"
 	"html/template"
 	"log"
 	"net/http"
 
 	"github.com/gorilla/mux"
+	"golang.org/x/crypto/bcrypt"
 )
 
 type Server struct {
-	r *mux.Router
+	r  *mux.Router
+	db *sql.DB
 }
 
-func New() *Server {
+func New(db *sql.DB) *Server {
 	return &Server{
-		r: mux.NewRouter(),
+		r:  mux.NewRouter(),
+		db: db,
 	}
 }
 
 func (server *Server) Run() error {
```

Comment:
I know it's not part of the current PR, but I just wanted to give an advice. When a receiver variable is declared in a receiver function, it's a good practice to keep the name short (in this case it would be `s`) because it's clear to what it's referring to. Same goes for using the `i` variable in for loops or `err` for errors. If the intent behind the variable is not clear from the context or variable type, than it's good to use full names (the `HashPassword` function parameter is a good example; it's type is a string, so it's fine to call the variable `password` instead of just `p`).

What's also important to consider is the scope of the variable. For example, the `s` for `Server` would just be used in the scope of that function (same goes for the already mentioned `i` or `err`), but for example struct fields would have more descriptive names, because they are probably used on multiple places in the code, so you want to have proper names (also imagine reading `s.sm.cu.fn`, not really clear what it is, right? `s.sessionManager.currentUser.firstName` is a bit better).